estuary/ui
# Enable display of a collection specification on the details page

## Issues

The issues directly below are completely resolved by this PR:
https://github.com/estuary/ui/issues/1069
https://github.com/estuary/ui/issues/865

## Changes

### 865

The following features are included in this PR:

* Add a toggle button to the _Spec_ tab of a collection _Details_ page that allows the user to view a tabular or raw code representation of the specification.

* Add a loading state to the content of the _Spec_ tab of a collection _Details_ page.

## Tests

### Manually tested

Approaches to testing are as follows:

* Validate that the toggle button appears and functions as expected on the _Spec_ tab of a collection _Details_ page.

* Validate that the _Spec_ tab of a non-collection _Details_ page remains unchanged.

* Validate that the loading states of nearly all tables that use the `EntityTableBody` component remain unchanged. The ops logs table was **not** reviewed.

### Automated tests

N/A

## Screenshots

**_Specification_ tab of collection _Details_ page | Table view**

<img width="998" alt="pr_screenshot-1067-collection_spec_presentation-light-table" src="https://github.com/estuary/ui/assets/77648584/943b29f3-ea09-4c33-a5fa-05f81ff3d2a5">

<br />
<br />

**_Specification_ tab of collection _Details_ page | Table view | Loading**

<img width="996" alt="pr_screenshot-1067-collection_spec_presentation-light-table-loading" src="https://github.com/estuary/ui/assets/77648584/0a9a104a-d42b-4eaf-896d-bde07c67ac96">

<br />
<br />

**_Specification_ tab of collection _Details_ page | Editor view**

<img width="1000" alt="pr_screenshot-1067-collection_spec_presentation-light-code" src="https://github.com/estuary/ui/assets/77648584/5ebde10f-1eaf-43c0-a215-98b0025ea627">

<br />
<br />

**_Specification_ tab of collection _Details_ page | Editor view | Loading**

<img width="995" alt="pr_screenshot-1067-collection_spec_presentation-light-code-loading" src="https://github.com/estuary/ui/assets/77648584/fb756ed0-4d96-4ecb-aec5-ddc0cf878fc0">
